src/pages/home/sidebar/SidebarScreen.js

```diff
@@ -54,6 +55,21 @@ class SidebarScreen extends Component {
         Timing.start(CONST.TIMING.SIDEBAR_LOADED, true);
     }
 
+    /**
+     * Method called when the New Workspace menu item is selected
+     *
+     * We create a policy and open the Workspace card modal.
+     * We also make a call to refresh the cached policies in Onyx,
+     * followed by an update to the state of this menu.
+     */
+    onNewWorkspaceSelected() {
+        create();
+        getPolicyList();
+        this.setState({
+            hasWorkspacesAlready: getWorkspaceCount(),
+        });
+    }
```

Comment:
When we call `create()`, this sends a request to our API to create the new policy, but there is not waiting for it before calling `getPolicyList()` or `getWorkspaceCount()`.

If you put

```
console.log('count', getWorkspaceCount(), this.state.hasWorkspacesAlready);
```

right after `getPolicyList()`, it prints `count 0 0`, so new policy is not there to affect the count yet. My guess is that `getPolicyList()` is not really being useful either because it is going to try to get the policies from the API, but potentially the new policy has not been created yet.

I did the exercise of removing the lines:

```
getPolicyList();
this.setState({
    hasWorkspacesAlready: getWorkspaceCount(),
});
```
and the behaviour seems to be the same with or without them. `hasWorkspacesAlready` gets recalculated each time we open the `Create Menu`.

Reply:
Makes sense, since the recalculation happens every time we open the menu. I'll go ahead and remove those lines! 👍